public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -350,7 +386,26 @@ function isScopeObj(obj: unknown): obj is Scope {
   );
 }
 
-function parseScopesFromLocalStorage(content: string | undefined): Scope[][] {
+function hasValidScopeParentNode(obj: unknown): obj is RecentScope {
+  return (
```

Comment:
I was a bit lazy when creating the other, type check, but maybe we can just use Zod here? We already have it as dependency and it's a nice library for doing this kind of runtime checks.